src/screens/register/index.js

```diff
@@ -1,182 +1,64 @@
-import StepThree from './step-three-code'
-import StepTwo from './step-two-master-pass'
-import StepOne from './step-one'
+import Success from './success'
+import Steps from './steps'
 
-function getStepContent(stepIndex) {
-  switch (stepIndex) {
-    case 0:
-      return 'Select your username and password.'
-    case 1:
-      return 'Select your Master Password'
-    case 2:
-      return 'This is your recovery code. You should really write this down.'
-    default:
-      return 'Uknown stepIndex'
-  }
-}
+const Register = () => (
+  <Formik
+    initialValues={{email: '', name: '', password: '', totpCode: ''}}
+    validate={values => {
+      const errors = {}
 
-class Register extends React.Component {
-  state = {
-    activeStep: 0
-  }
+      if (!values.email) {
+        errors.email = 'Required'
+      } else if (!/@/.test(values.email)) {
+        errors.email = 'Invalid email address'
+      } else if (values.email.length > 255) {
+        errors.email = 'Has more than 255 characters'
+      }
 
-  getSteps = () => {
-    return [
-      <StepOne key={0} props={this.state} />,
-      <StepTwo key={1} props={this.state} />,
-      <StepThree key={2} props={this.state} />,
-      <StepFour key={3} props={this.state} />,
-      <StepFive key={4} props={this.state} />
-    ]
-  }
+      if (!values.name) {
+        errors.name = 'Required'
+      } else if (values.name.length > 255) {
+        errors.name = 'Has more than 255 characters'
+      }
 
-  getCurrentStep = () => {
-    if (this.state.activeStep === 0) {
-      return [<StepOne key={0} props={this.state} />]
-    }
-    if (this.state.activeStep === 1) {
-      return [<StepTwo key={0} props={this.state} />]
-    }
-    if (this.state.activeStep === 2) {
-      return [<StepThree key={0} props={this.state} />]
-    }
-    if (this.state.activeStep === 3) {
-      return [<StepFour key={0} props={this.state} />]
-    }
-    if (this.state.activeStep === 4) {
-      return [<StepFive key={0} props={this.state} />]
-    }
-  }
+      if (!values.password) {
+        errors.password = 'Required'
+      } else if (values.password.length < 10) {
+        errors.password = 'Has less than 10 characters'
+      } else if (values.password.length > 255) {
+        errors.password = 'Has more than 255 characters'
+      }
 
-  handleNext = () => {
-    this.setState(state => ({
-      activeStep: state.activeStep + 1
-    }))
-  }
+      if (!values.totpCode) {
+        errors.totpCode = 'Required'
+      } else if (values.totpCode.length !== 6) {
+        errors.totpCode = 'Invalid code'
+      }
 
-  handleBack = () => {
-    this.setState(state => ({
-      activeStep: state.activeStep - 1
-    }))
-  }
+      return errors
+    }}
+    onSubmit={(values, {setSubmitting, setStatus}) => {
+      setTimeout(() => {
+        console.log(values)
+        setStatus({submitted: true})
+        setSubmitting(false)
+      }, 400)
+    }}
+  >
+    {props => {
```

Comment:
from where is this props getting its values? i noticed it isnt being sent to the component as an injected dependency, it just appears here in the middle of the code. Does formik automatically provides this object with the form values typed so far by the user?

Reply:
@gugutz this props come from formik. It used the render prop technique to pass to it's children the props. This props contain the values, handleChange etc. https://reactjs.org/docs/render-props.html